cm-vault/vault-config/variables.tf

```diff
@@ -2,3 +2,8 @@ variable "vault_pki_root_domain" {
   type    = string
   default = "cert-manager.io"
 }
+
+variable "vault_authentication_mechanism" {
+  type    = string
+  default = "approle"
+}
```

Comment:
Suggestion: validate that the input value is either 'approle' or 'token'. I think you could use Terraform's custom validtion for this https://www.terraform.io/language/values/variables#custom-validation-rules

Reply:
Thanks! I guess it would be good if it was possible to validate these values at the top, potentially with Terragrunt. At the moment if incorrect value is passed, only the vault-config module creation fails, the other continue to be applied whereas what we actually want is for the whole thing to exit as soon as possible. Perhaps we can look at that after #23